services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/PaginatedQueryInterface.java

```diff
@@ -0,0 +1,67 @@
+package us.dot.its.jpo.ode.api.accessors;
+
+import org.bson.Document;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.*;
+import org.springframework.data.mongodb.core.query.Criteria;
+import us.dot.its.jpo.ode.api.models.AggregationMetadata;
+import us.dot.its.jpo.ode.api.models.AggregationResult;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public interface PaginatedQueryInterface {
+    /**
+     * Find paginated data based on the given criteria and pageable object
+     *
+     * @param <T>            the type of the entity to return
+     * @param mongoTemplate  the mongo template object to query with
+     * @param collectionName the collection name to query
+     * @param pageable       the pageable object to use for pagination
+     * @param criteria       the criteria object to use for querying
+     * @param sort           the sort object to use for sorting
+     * @return the paginated data that matches the given criteria
+     */
+    default <T> Page<T> findPaginatedData(
```

Comment:
suggestion(non-blocking): the `data` suffix is already implied by the fact we're trying to "find" something. We could simplify to 
```suggestion
    default <T> Page<T> findPage(
```
or simply
```suggestion
    default <T> Page<T> findAll(
```

Reply:
updated